pkg/iac/scanners/kubernetes/scanner_test.go

```diff
@@ -336,6 +336,339 @@ deny[res] {
+ }
+
+ added_capability(container, cap) {
+ 	container.securityContext.capabilities.add[_] == cap
+ }
+
+ has_field(obj, field) {
+ 	obj[field]
+ }
+
+ no_read_only_filesystem(c) {
+ 	not has_field(c, "securityContext")
+ }
+
+ no_read_only_filesystem(c) {
+ 	has_field(c, "securityContext")
+ 	not has_field(c.securityContext, "readOnlyRootFilesystem")
+ }
+
+ privilege_escalation_allowed(c) {
+ 	not has_field(c, "securityContext")
+ }
+
+ privilege_escalation_allowed(c) {
+ 	has_field(c, "securityContext")
+ 	has_field(c.securityContext, "allowPrivilegeEscalation")
+ }
+
+ annotations[annotation] {
+ 	pods[pod]
+ 	annotation = pod.metadata.annotations
+ }
+
+ host_ipcs[host_ipc] {
+ 	pods[pod]
+ 	host_ipc = pod.spec.hostIPC
+ }
+
+ host_networks[host_network] {
+ 	pods[pod]
+ 	host_network = pod.spec.hostNetwork
+ }
+
+ host_pids[host_pid] {
+ 	pods[pod]
+ 	host_pid = pod.spec.hostPID
+ }
+
+ host_aliases[host_alias] {
+ 	pods[pod]
+ 	host_alias = pod.spec
+ }
+ `,
+		"/rules/lib.util.rego": `
+ package lib.utils
+
+ has_key(x, k) {
+ 	_ = x[k]
+ }`,
+		"/rules/rule.rego": `
+package builtin.kubernetes.KSV011
+
+import data.lib.kubernetes
+import data.lib.utils
+
+default failLimitsCPU = false
+
+__rego_metadata__ := {
+	"id": "KSV011",
+	"avd_id": "AVD-KSV-0011",
+	"title": "CPU not limited",
+	"short_code": "limit-cpu",
+	"version": "v1.0.0",
+	"severity": "LOW",
+	"type": "Kubernetes Security Check",
+	"description": "Enforcing CPU limits prevents DoS via resource exhaustion.",
+	"recommended_actions": "Set a limit value under 'containers[].resources.limits.cpu'.",
+	"url": "https://cloud.google.com/blog/products/containers-kubernetes/kubernetes-best-practices-resource-requests-and-limits",
+}
+
+__rego_input__ := {
+	"combine": false,
+	"selector": [{"type": "kubernetes"}],
+}
+
+# getLimitsCPUContainers returns all containers which have set resources.limits.cpu
+getLimitsCPUContainers[container] {
+	allContainers := kubernetes.containers[_]
+	utils.has_key(allContainers.resources.limits, "cpu")
+	container := allContainers.name
+}
+
+# getNoLimitsCPUContainers returns all containers which have not set
+# resources.limits.cpu
+getNoLimitsCPUContainers[container] {
+	container := kubernetes.containers[_].name
+	not getLimitsCPUContainers[container]
+}
+
+# failLimitsCPU is true if containers[].resources.limits.cpu is not set
+# for ANY container
+failLimitsCPU {
+	count(getNoLimitsCPUContainers) > 0
+}
+
+deny[res] {
+	failLimitsCPU
+
+	msg := kubernetes.format(sprintf("Container '%s' of %s '%s' should set 'resources.limits.cpu'", [getNoLimitsCPUContainers[_], kubernetes.kind, kubernetes.name]))
+
+	res := {
+		"msg": msg,
+		"id": __rego_metadata__.id,
+		"title": __rego_metadata__.title,
+		"severity": __rego_metadata__.severity,
+		"type": __rego_metadata__.type,
+        "startline": 6,
+        "endline": 10,
+	}
+}
+`,
+	})
+
+	scanner := NewScanner(options.ScannerWithPolicyDirs("rules"))
```

Comment:
You can load the libraries and then you can remove the library functions from Rego. The same can be done for the test with YAML.
```suggestion
	scanner := NewScanner(
		options.ScannerWithPolicyDirs("rules"),
		options.ScannerWithEmbeddedLibraries(true),
	)
```

Reply:
done https://github.com/aquasecurity/trivy/pull/6490/commits/2e31e2a5e8e0db4916b331983f4803a46ec7caf9